tests/numerics/fv/test_tpsa.py

```diff
@@ -21,1418 +36,1426 @@
+        case "dir":
+            bc_disp = pp.BoundaryConditionVectorial(
+                grid, faces=face_ind, cond=nf * ["dir"]
+            )
+        case "neu":
+            bc_disp = pp.BoundaryConditionVectorial(
+                grid, faces=face_ind, cond=nf * ["neu"]
+            )
+        case "rob":
+            bc_disp = pp.BoundaryConditionVectorial(
+                grid, faces=face_ind, cond=nf * ["rob"]
+            )
+        case _:
+            raise ValueError(f"Unknown boundary condition type {bc_type}")
 
-        # The contribution from cell center displacement to the boundary displacement
-        # has unit value in the cell neighboring the face.
-        bound_displacement_cell = np.zeros((4, 4))
-        bound_displacement_cell[1, 1] = 1
-        bound_displacement_cell[3, 3] = 1
+    d[pp.PARAMETERS][KEYWORD]["bc"] = bc_disp
 
-        # On Dirichlet faces, the boundary displacement is recovered from the boundary
-        # condition; all other reconstruction matrices are zero.
-        bound_displacement_face = np.zeros((4, 14))
-        bound_displacement_face[0, 0] = 1
-        bound_displacement_face[1, 1] = -self.d_0_0 / (2 * self.mu_0 * self.n_0_nrm)
-        bound_displacement_face[2, 12] = 1
-        bound_displacement_face[3, 13] = self.d_1_6 / (2 * self.mu_1 * self.n_6_nrm)
 
-        # The contribution from solid pressure. Multiply with -1 on face 0 since this
-        # has an inward pointing normal vector.
-        bound_displacement_rotation_cell = np.zeros((4, 2))
-        bound_displacement_rotation_cell[1, 0] = (
-            -self.d_0_0 * self.n_0[0] / (2 * self.mu_0 * self.n_0_nrm)
-        )
-        bound_displacement_rotation_cell[3, 1] = (
-            self.d_1_6 * self.n_6[0] / (2 * self.mu_1 * self.n_6_nrm)
-        )
+def _set_bc_by_direction(
+    g: pp.Grid,
+    d: dict,
+    type_south: Literal["dir", "neu"],
+    type_east: Literal["dir", "neu"],
+    type_north: Literal["dir", "neu"],
+    type_west: Literal["dir", "neu"],
+    type_bottom: Optional[Literal["dir", "neu"]] = None,
+    type_top: Optional[Literal["dir", "neu"]] = None,
+) -> np.ndarray:
+    """Set the boundary conditions on the grid, based on the types of boundary
+    conditions given.
 
-        # Contribution from solid pressure. Multiply with -1 on face 0 since this has an
-        # inward pointing normal vector.
-        bound_displacement_solid_pressure_cell = np.zeros((4, 2))
-        bound_displacement_solid_pressure_cell[1, 0] = (
-            -self.d_0_0 * self.n_0[1] / (2 * self.mu_0 * self.n_0_nrm)
-        )
-        bound_displacement_solid_pressure_cell[3, 1] = (
-            self.d_1_6 * self.n_6[1] / (2 * self.mu_1 * self.n_6_nrm)
-        )
+    The boundary condition has value 0.1 on the south face, -0.1 on the east face, and
+    0.1 on the bottom face (if 3d). The remaining faces have homogeneous conditions.
+    The conditions correspond to a compressive force on the south, east and (if
```

Comment:
Whether the condition is compressive or tensile depends on the bc values, no?

Reply:
This function sets a compressive regime, while tensile can be achieved by negation in the calling function.This is now explicitly noted (last line of this paragraph).